### ISSUE TITLE
[CS2] Quotation mark characters are escaped unnecessarily in template literals

### ISSUE THREAD

**Issue by Inve1951** [jashkenas/coffeescript#4589]
```COFFEE
"""<div id="unique" onclick="update('#{someVar}')"></div>"""
```
[compiles to](http://coffeescript.org/v2/#try:%22%22%22%3Cdiv%20id%3D%22unique%22%20onclick%3D%22update('%23%7BsomeVar%7D')%22%3E%3C%2Fdiv%3E%22%22%22)
```JS
`<div id=\"unique\" onclick=\"update('${someVar}')\"></div>`;
```

`\"` doesn't need escaping

**Comment by GeoffreyBooth** [comment]
Agreed that the `"`s don’t need to be escaped in the second example, but what’s wrong with the first? `#{` isn’t supposed to interpolate in `'`-delimited strings.

**Comment by Inve1951** [comment]
> `#{` isn’t supposed to interpolate in '-delimited strings.

it's not? i thought `'` and `"` are interchangeable

**Comment by GeoffreyBooth** [comment]
>  i thought `'` and `"` are interchangeable

Nope: http://coffeescript.org/v2/#strings. It’s always been that way, that’s not a v2 thing. CoffeeScript is mimicking Ruby and Bash (among others) in this regard.

**Comment by Inve1951** [comment]
i see. want to close and open a new issue or leave it?

**Comment by GeoffreyBooth** [comment]
Up to you. Maybe just edit your first post?

**Comment by GeoffreyBooth** [comment]
Fixed by #4660.
